echopop/utils/batch_load.py

```diff
@@ -0,0 +1,608 @@
+import glob
+import os
+import re
+from pathlib import Path
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from ..core import CONFIG_MAP, ECHOVIEW_EXPORT_MAP, REGION_EXPORT_MAP
+from ..spatial.transect import export_transect_layers, export_transect_spacing
+from .operations import group_merge
+
+
+def read_echoview_export(filename: str, transect_number: int):
+    """
+    Read Echoview export file
+
+    Parameters
+    ----------
+    filename: str
+        Export file path.
+    transect_number: int
+        Transect number associated with the export file.
+
+    See Also
+    ----------
+    echopop.load.batch_read_echoview_exports
+        A more detailed description.
+    """
+
+    # Read csv file
+    export_file = pd.read_csv(filename, index_col=None, header=0, skipinitialspace=True)
+
+    # Assign transect number column
+    export_file["transect_num"] = transect_number
+
+    # Get coordinate column names, if present
+    # ---- Extract longitude column name
+    lon_col = next((col for col in export_file.columns if "lon" in col.lower()), None)
+    # ---- Extract latitude column name
+    lat_col = next((col for col in export_file.columns if "lat" in col.lower()), None)
+
+    # Automatic "bad data" removal/imputation for georeferenced (Latitude/Longitude) data
+    # ---- Helper imputation function
+    def impute_bad_data(data, column):
```

Comment:
Is there a reason why you don't want to use [pd.dataframe.fillna](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.fillna.html) for these?

Reply:
I have tried implementing this by created an appropriately second indexed `DataFrame` to fill in the values, but that didn't seem to match what Chu's code produced. The unfortunate thing here is that these are rolling imputations, so they are conditional on the previously imputed values. I guess an alternative would be to try and compute the slopes between coordinates and impute that way; however, I tried doing that previously and it didn't seem to work out. That could have very well been simply due to an indexing issue on my end.